lib/Doctrine/DBAL/Types/DecimalType.php

```diff
@@ -49,6 +49,6 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
      */
     public function convertToPHPValue($value, AbstractPlatform $platform)
     {
-        return (null === $value) ? null : $value;
+        return (null === $value) ? null : (string) $value;
```

Comment:
You should fix this in the driver for now. It is a known problem that we do not like anything else then strings. 
I think beberlei have some plans to fix it in the future. If we lucky someone will do it when they get annoyed enough by it :)